### PR TITLE
Extract and test manual examples

### PR DESCRIPTION
**Pull request by olexandr-konovalov**
Follows setup seen in https://github.com/gap-packages/orb/pull/46/.

### CODE REVIEW

**Comment by olexandr-konovalov** [comment]
Clearly tests fail - need now to update examples, mainly in Chapter 3.

File: lib/ybe.gd [hunk 1]
```diff
@@ -99,8 +99,7 @@ DeclareAttribute("DerivedRightRack", IsYB);
 #! @ExampleSession
 #! gap> cs := SmallCycleSet(4,13);;
 #! gap> yb := CycleSet2YB(cs);;
-#| gap> DehornoyClass(yb);
-#! 2
+#| gap> DehornoyClass(yb);;
```
Comment (fingolfin) [inline lib/ybe.gd]: Alternatively, you could keep the first line as-is, and just change the second line to `#| 2`

Comment (olexandr-konovalov) [reply lib/ybe.gd]: That's the least problem. Some examples do not work and trigger a recursion depth trap. @vendramin we need to investigate.

**Comment by olexandr-konovalov** [comment]
Now all examples pass, except one. It catches a recursion depth trap with assertion level 2:
```
gap> SetAssertionLevel(2);
gap> LoadPackage("Yan");
─────────────────────────────────────────────────────────────────────────────
Loading  YangBaxter 0.8.0 (Combinatorial Solutions for the Yang-Baxter equation)
by Leandro Vendramin (http://mate.dm.uba.ar/~lvendram) and
   Alexander Konovalov (https://alexk.host.cs.st-andrews.ac.uk/).
Homepage: https://gap-packages.github.io/YangBaxter
─────────────────────────────────────────────────────────────────────────────
true
gap> IdBrace(IYBBrace(SmallIYB(5,86)));
Error, recursion depth trap (5000) in
  if IsInt( p ) and IsPrimeInt( p ) then
    subfield := p;
    if IsInt( d ) and 0 < d then
        if MAXSIZE_GF_INTERNAL < p ^ d then
            return LargeGaloisField( p, d );
        fi;
    elif
     IsRationalFunction( d ) and IsLaurentPolynomial( d ) 
        and DegreeFFE( CoefficientsOfLaurentPolynomial( d )[1] ) = 1 then
        return FieldExtension( GaloisField( p, 1 ), d );
    elif IsHomogeneousList( d ) and DegreeFFE( d ) = 1 then
        return FieldExtension( GaloisField( p, 1 ), 
           UnivariatePolynomial( GaloisField( p, 1 ), d ) );
    elif IsHomogeneousList( d ) then
        F := GaloisField( GaloisField( p, 1 ), Length( d ) );
        B := Basis( F, d );
        if B = fail then
            Error( "<extension> is not linearly independent" );
        fi;
        SetBasis( F, B );
        return F;
    fi;
elif IsField( p ) then
    subfield := p;
    p := Characteristic( subfield );
    d1 := DegreeOverPrimeField( subfield );
    if IsInt( d ) then
        q := p ^ (d * d1);
        if MAXSIZE_GF_INTERNAL < q then
            if d1 = 1 then
                return LargeGaloisField( p, d );
            else
                return FieldByGenerators( subfield, [ Z( p, d * d1 ) ] );
            fi;
        fi;
        d := d * DegreeOverPrimeField( subfield );
    elif
     IsHomogeneousList( d ) 
        and DegreeOverPrimeField( subfield ) mod DegreeFFE( d ) = 0 then
        return FieldExtension( subfield, UnivariatePolynomial( subfield, d ) )
         ;
    elif
     IsRationalFunction( d ) and IsLaurentPolynomial( d ) 
        and DegreeOverPrimeField( subfield ) 
            mod DegreeFFE( CoefficientsOfLaurentPolynomial( d )[1] ) = 0 then
        return FieldExtension( subfield, d );
    elif IsHomogeneousList( d ) then
        F := GaloisField( subfield, Length( d ) );
        B := Basis( F, d );
        if B = fail then
            Error( "<extension> is not linearly independent" );
        fi;
        SetBasis( F, B );
        return F;
    else
        Error( "<extension> must be a <deg>, <bas>, or <pol>" );
    fi;
else
    Error( "<subfield> must be a prime or a finite field" );
fi; at /usr/local/Cellar/gap/4.10.2/libexec/lib/ffe.gi:467 called from 
GF( Characteristic( gens ), DegreeFFE( gens ) 
 ) at /usr/local/Cellar/gap/4.10.2/libexec/lib/ffe.gi:1072 called from
DefaultRingByGenerators( arg[1] 
 ) at /usr/local/Cellar/gap/4.10.2/libexec/lib/ring.gi:412 called from
DefaultRing( coeff ) at /usr/local/Cellar/gap/4.10.2/libexec/lib/ratfun1.gi:47 called from
LaurentPolynomialByExtRepNC( fam, cofs, val, ind 
 ) at /usr/local/Cellar/gap/4.10.2/libexec/lib/ratfunul.gi:43 called from
LaurentPolynomialByCoefficients( FamilyObj( c[1] ), c, 0, inum 
 ) at /usr/local/Cellar/gap/4.10.2/libexec/lib/ratfunul.gi:890 called from
...  at *stdin*:3
you may 'return;'
brk> 
```

**Comment by olexandr-konovalov** [comment]
the other end of the call stack is
```
at /usr/local/Cellar/gap/4.10.2/libexec/lib/ghom.gi:309 called from
GroupGeneralMappingByImagesNC( G, H, gens, imgs ) at /usr/local/Cellar/gap/4.10.2/libexec/lib/ghom.gi:439 called from
GroupHomomorphismByImagesNC( N, S, perms, Sgens ) at /usr/local/Cellar/gap/4.10.2/libexec/pkg/cryst/gap/cryst.gi:592 called from
PointGroupHomomorphism( S ) at /usr/local/Cellar/gap/4.10.2/libexec/pkg/cryst/gap/cryst.gi:609 called from
PointGroup( S ) at /usr/local/Cellar/gap/4.10.2/libexec/pkg/cryst/gap/cryst.gi:499 called from
x in H at /usr/local/Cellar/gap/4.10.2/libexec/lib/ghom.gi:309 called from
func( elm ) at /usr/local/Cellar/gap/4.10.2/libexec/lib/coll.gi:1530 called from
ForAll( imgs, function ( x )
      return x in H;
  end ) at /usr/local/Cellar/gap/4.10.2/libexec/lib/ghom.gi:309 called from
GroupGeneralMappingByImagesNC( G, H, gens, imgs ) at /usr/local/Cellar/gap/4.10.2/libexec/lib/ghom.gi:439 called from
GroupHomomorphismByImagesNC( N, S, perms, Sgens ) at /usr/local/Cellar/gap/4.10.2/libexec/pkg/cryst/gap/cryst.gi:592 called from
PointGroupHomomorphism( S ) at /usr/local/Cellar/gap/4.10.2/libexec/pkg/cryst/gap/cryst.gi:609 called from
PointGroup( S ) at /usr/local/Cellar/gap/4.10.2/libexec/pkg/cryst/gap/cryst.gi:155 called from
TranslationBasisFun( S ) at /usr/local/Cellar/gap/4.10.2/libexec/pkg/cryst/gap/cryst.gi:222 called from
TranslationBasis( TransposedMatrixGroup( S ) ) at /usr/local/Cellar/gap/4.10.2/libexec/pkg/cryst/gap/cryst.gi:224 called from
TranslationBasis( S ) at /usr/local/Cellar/gap/4.10.2/libexec/pkg/cryst/gap/cryst.gi:517 called from
x in group at /Users/alexk/Library/Preferences/GAP/pkg/YangBaxter/lib/ybe.gi:681 called from
func( elm ) at /usr/local/Cellar/gap/4.10.2/libexec/lib/list.gi:2790 called from
First( AllMatrices( hol, v ), function ( x )
      return x in group;
  end ) at /Users/alexk/Library/Preferences/GAP/pkg/YangBaxter/lib/ybe.gi:681 called from
ESS_inverse1cocycle( a + b ) at /Users/alexk/Library/Preferences/GAP/pkg/YangBaxter/lib/ybe.gi:695 called from
sum( x, y ) at /Users/alexk/Library/Preferences/GAP/pkg/YangBaxter/lib/ybe.gi:704 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *errin*:2
brk> 
```

**Comment by olexandr-konovalov** [comment]
One more thing - for stable-4.9 only there is a warning: 
```
########> Diff in /home/travis/build/gap-packages/YangBaxter/gaproot/pkg/YangB\
axter/tst/testextra/example32.tst:4
# Input is:
SortedList(List(ideals, IdBrace));
# Expected output:
[ [ 1, 1 ], [ 2, 1 ], [ 4, 1 ], [ 4, 2 ], [ 4, 3 ], [ 8, 10 ], [ 8, 13 ], 
  [ 8, 19 ], [ 16, 73 ] ]
# But found:
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
#I  The group tested requires many generators. `AllHomomorphismClasses' often
#I  does not perform well for such groups -- see the documentation.
[ [ 1, 1 ], [ 2, 1 ], [ 4, 1 ], [ 4, 2 ], [ 4, 3 ], [ 8, 10 ], [ 8, 13 ], 
  [ 8, 19 ], [ 16, 73 ] ]
########
```
Perhaps it points to some inefficiencies in the code?

**Comment by olexandr-konovalov** [comment]
The latter warning is produced by `AllHomomorphismClasses`:
```
  if IsFinite(H) and Length(bg)>2 and cnt>Size(H)^2 and Size(G)<bw then
    Info(InfoPerformance,1,
"The group tested requires many generators. `AllHomomorphismClasses' often\n",
"#I  does not perform well for such groups -- see the documentation.");
  fi;
```
and it's present only in GAP 4.9 - perhaps some improvements in later versions lead to some optimisations, so the condition does not hold. I can silence the warning by temporarily decreasing `InfoPerformance`.

**Comment by olexandr-konovalov** [comment]
@vendramin this is merged. We have now all but one manual examples under regression testing. Test files with extracted manual examples are kept under version control - to update them, you need to run `makedoc.g` in GAP.

**Pull request merged**